### PR TITLE
Converted store/storetest/compliance_store.go t.Fatal calls into asse…

### LINKED ISSUE #12784: Migrate tests from "store/storetest/compliance_store.go" to use testify
Mattermost is migrating its server tests from using the `t.Fatal` method calls on their checks to use the [testify toolkit](https://github.com/stretchr/testify). The goal of this migration is to improve the readability of the tests, making them more concise and meaningful. This Help Wanted issue is to migrate the tests in the `store/storetest/compliance_store.go` file.

The expected way to migrate this is to go to the `store/storetest/compliance_store.go` file and modify the calls to `t.Fatal` and the condition checks that lead to them, replacing them with calls to the [the `require` package](https://godoc.org/github.com/stretchr/testify/require) if a failure should stop the execution of the test, or [the `assert` package](https://godoc.org/github.com/stretchr/testify/assert) if it shouldn't.

Example: mattermost/mattermost-server#12039

If you have questions about the ticket or you need any help, feel free to contact `miguel.delacruz` or `jesus.espino` in <https://community.mattermost.com/>

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-19452

### PR DESCRIPTION
**Pull request by rv404674**
#### Summary
Converted store/storetest/compliance_store.go t.Fatal calls into assert/require calls.

#### Ticket Link
Fixes #12784

### CODE REVIEW

File: store/storetest/compliance_store.go [hunk 1]
```diff
@@ -54,9 +54,7 @@ func testComplianceStore(t *testing.T, ss store.Store) {
 
 	compliances, _ = ss.Compliance().GetAll(1, 1)
 
-	if len(compliances) != 1 {
-		t.Fatal("should only have returned 1")
-	}
+	require.Equal(t, 1, len(compliances), "should only have returned 1")
```
Comment (mgdelacroix) [inline store/storetest/compliance_store.go]: You can use `require.Len` here instead of `require.Equal` to improve readability

**Comment by hanzei** [comment]
Hi @rv404674,

Is the provided feedback clear? Let me know, if I can help with anything.

**Comment by rv404674** [comment]
Hi @hanzei . Yup the feedback is clear. I am out of station right now, will it be fine If I complete it in a week??

Thanks.

**Comment by hanzei** [comment]
Sure @rv404674. I just wanted to make sure you are not blocked from our side.

**Comment by jasonblais** [comment]
Hi @rv404674, let us know if you're still working on this PR? We'd be happy to answer any questions you may have.

Note that after 10 days, we'll close the PR if there is no further activity and open the pull request for other community members to work on.

**Comment by hanzei** [comment]
Hi @rv404674,

I'm going to close this PR due to inactivity. Let me know, if you still want to work on this.

**Pull request closed without merging**